sdk/src/user.ts

```diff
@@ -1821,35 +1956,113 @@ export class User {
 		canBeLiquidated: boolean;
 		marginRequirement: BN;
 		totalCollateral: BN;
+		liquidationStatuses: Map<
+			'cross' | number,
+			{ canBeLiquidated: boolean; marginRequirement: BN; totalCollateral: BN }
+		>;
 	} {
-		const liquidationBuffer = this.getLiquidationBuffer();
+		// Deprecated signature retained for backward compatibility in type only
+		// but implementation now delegates to the new Map-based API and returns cross margin status.
+		const map = this.getLiquidationStatuses();
+		const cross = map.get('cross');
+		return cross
+			? { ...cross, liquidationStatuses: map }
+			: {
+					canBeLiquidated: false,
+					marginRequirement: ZERO,
+					totalCollateral: ZERO,
+					liquidationStatuses: map,
+			  };
+	}
 
-		const totalCollateral = this.getTotalCollateral(
-			'Maintenance',
-			undefined,
-			undefined,
-			liquidationBuffer
-		);
+	/**
+	 * New API: Returns liquidation status for cross and each isolated perp position.
+	 * Map keys:
+	 *  - 'cross' for cross margin
+	 *  - marketIndex (number) for each isolated perp position
+	 */
+	public getLiquidationStatuses(
+		marginCalc?: MarginCalculation
+	): Map<
+		'cross' | number,
+		{ canBeLiquidated: boolean; marginRequirement: BN; totalCollateral: BN }
+	> {
+		// If not provided, use buffer-aware calc for canBeLiquidated checks
+		if (!marginCalc) {
+			const liquidationBuffer = this.getLiquidationBuffer();
+			marginCalc = this.getMarginCalculation('Maintenance', {
+				liquidationBuffer,
+			});
+		}
 
-		const marginRequirement =
-			this.getMaintenanceMarginRequirement(liquidationBuffer);
-		const canBeLiquidated = totalCollateral.lt(marginRequirement);
+		const result = new Map<
+			'cross' | number,
+			{
+				canBeLiquidated: boolean;
+				marginRequirement: BN;
+				totalCollateral: BN;
+			}
+		>();
+
+		// Cross margin status
+		const crossTotalCollateral = marginCalc.totalCollateral;
+		const crossMarginRequirement = marginCalc.marginRequirement;
+		result.set('cross', {
+			canBeLiquidated: crossTotalCollateral.lt(crossMarginRequirement),
+			marginRequirement: crossMarginRequirement,
+			totalCollateral: crossTotalCollateral,
+		});
 
-		return {
-			canBeLiquidated,
-			marginRequirement,
-			totalCollateral,
-		};
+		// Isolated positions status
+		for (const [
+			marketIndex,
+			isoCalc,
+		] of marginCalc.isolatedMarginCalculations) {
+			const isoTotalCollateral = isoCalc.totalCollateral;
+			const isoMarginRequirement = isoCalc.marginRequirement;
+			result.set(marketIndex, {
+				canBeLiquidated: isoTotalCollateral.lt(isoMarginRequirement),
+				marginRequirement: isoMarginRequirement,
+				totalCollateral: isoTotalCollateral,
+			});
+		}
+
+		return result;
 	}
 
-	public isBeingLiquidated(): boolean {
-		return (
+	public isBeingLiquidated(marginCalc?: MarginCalculation): boolean {
+		// Consider on-chain flags OR computed margin status (cross or any isolated)
```

Comment:
idk if we should change this

Reply:
Ok as in just use the on-chain flag only?